src/main.cpp

```diff
@@ -3219,16 +3207,19 @@ bool static ConnectTip(CValidationState &state,
         bool result = view.Flush();
         assert(result);
     }
+
     int64_t nTime4 = GetTimeMicros();
     nTimeFlush += nTime4 - nTime3;
     LogPrint("bench", "  - Flush: %.2fms [%.2fs]\n", (nTime4 - nTime3) * 0.001, nTimeFlush * 0.000001);
-    // Write the chain state to disk, if necessary.
-    if (!FlushStateToDisk(state, FLUSH_STATE_IF_NEEDED))
-        return false;
+    // Write the chain state to disk, if necessary, and only during IBD, reindex, or importing.
+    if (!IsChainNearlySyncd() || fReindex || fImporting)
+        if (!FlushStateToDisk(state, FLUSH_STATE_IF_NEEDED))
+            return false;
```

Comment:
There are two reasons:  1) We flush ALWAYS now after every new block is processed (when processnewblock returns, at the end of HandleBlockMessageThread) during normal operation so checking again is redundant, furthermore, it injects performance timing issues if we flush at that point during ConnectTip  it can add a significant amount of time to the processing of a block in our statistics, when acutally, it has nothing to do with block processsing and should be excluded from the timing.